quisp/modules/QRSA/HardwareMonitor/IHardwareMonitor.h

```diff
@@ -71,8 +71,10 @@ typedef std::map<int, InterfaceInfo> NeighborTable;
 // e.g.
 // "XX" -> {plus_plus = 56, plus_minus = 55, minus_plus = 50, minus_minus = 50},
 // "XY" -> {....
-typedef std::map<std::string, output_count> raw_data;
-typedef std::map<int, tomography_outcome> Temporal_Tomography_Output_Holder;  // measurement_count_id -> outcome. For single qnic
+typedef std::map<int, std::map<std::string, output_count>> raw_data;  // qnic -> partner . basis string . output
+typedef std::map<int, link_cost> link_cost_map;
+typedef std::map<int, tomography_outcome> TemporalTomographyOutputHolder;  // measurement_count_id -> outcome. For single qnic
+typedef std::map<int, std::map<int, tomography_outcome>> TomographyOutcome;  // partner -> <count_id, outcome>
```

Comment:
this naming is so confusing. `TomographyOutcome` is a `map<int, tomography_outcome>`? what's the difference between `TomographyOutcome` and `tomography_outcome` ? absolutely misleading name.

Reply:
Right. So how about `TomographyOutcomeTable`? This would be still confusing?

Reply:
it's a better name. but is that a table? hmm...